src/antares/service/local_services/thermal_local.py

```diff
@@ -31,7 +39,74 @@ def update_thermal_properties(
         raise NotImplementedError
 
     def get_thermal_matrix(self, thermal_cluster: ThermalCluster, ts_name: ThermalClusterMatrixName) -> pd.DataFrame:
-        raise NotImplementedError
+        if ts_name.value == "series":
+            time_serie_type = TimeSeriesFileType.THERMAL_SERIES
+        elif ts_name.value == "modulation":
+            time_serie_type = TimeSeriesFileType.THERMAL_MODULATION
+        elif ts_name.value == "data":
+            time_serie_type = TimeSeriesFileType.THERMAL_DATA
+        elif ts_name.value == "CO2Cost":
+            time_serie_type = TimeSeriesFileType.THERMAL_CO2
+        elif ts_name.value == "fuelCost":
+            time_serie_type = TimeSeriesFileType.THERMAL_FUEL
+
```

Comment:
You should either put a else: raise NotImplementedError here or a else with the fuelCost as here we can reach a null pointer as `time_serie_type` could be undefined